packages/babel-helpers/src/helpers/usingCtx.ts

```diff
@@ -0,0 +1,81 @@
+/* @minVersion 7.23.9 */
+
+type Stack = {
+  v: any;
+  d: () => any;
+  a: boolean;
+};
+
+export default function _usingCtx() {
+  var _disposeSuppressedError =
+      typeof SuppressedError === "function"
+        ? // eslint-disable-next-line no-undef
+          SuppressedError
+        : (function (error: Error, suppressed: Error) {
+            var err = new Error() as SuppressedError;
+            err.name = "SuppressedError";
+            err.suppressed = suppressed;
+            err.error = error;
+            return err;
+          } as SuppressedErrorConstructor),
+    empty = {},
+    stack: Stack[] = [];
+  function using(isAwait: boolean, value: any) {
+    if (value != null) {
+      if (Object(value) !== value) {
+        throw new TypeError(
+          "using declarations can only be used with objects, functions, null, or undefined.",
+        );
+      }
+      // core-js-pure uses Symbol.for for polyfilling well-known symbols
```

Comment:
`core-js-pure` seems to define `Symbol.asyncDispose` directly. 🤔
![image](https://github.com/babel/babel/assets/30521560/214522e7-96a4-49a5-af9d-f36109423f54)
![image](https://github.com/babel/babel/assets/30521560/bb864db5-8bde-48e2-9232-2d5e5416669e)
![image](https://github.com/babel/babel/assets/30521560/7e98f720-d20a-4cd6-8ab8-d31807da9c3f)

Reply:
Not necessarily, but maybe users are importing "things that can be disposed" from libraries that have been compiled with core-js, and that thus use `Symbol.for("Symbol.dispose")` as their protocol.